kubernetes/config/kube_config.c

```diff
@@ -393,8 +386,48 @@ int load_kube_config(char **pBasePath, sslConfig_t ** pSslConfig, list_t ** pApi
     }
 
   end:
+    return rc;
+}
+
+int load_kube_config(char **pBasePath, sslConfig_t ** pSslConfig, list_t ** pApiKeys, const char *configFileName)
+{
+    static char fname[] = "load_kube_config()";
+    int rc = 0;
+
+    kubeconfig_t *kubeconfig = kubeconfig_create();
+    if (!kubeconfig) {
+        fprintf(stderr, "%s: Cannot create kubeconfig.[%s]\n", fname, strerror(errno));
+        return -1;
+    }
+
+    kubeconfig->fileName = getWorkingConfigFile(configFileName);
+
+    rc = load_kube_config_common(pBasePath, pSslConfig, pApiKeys, kubeconfig);
+
+    kubeconfig_free(kubeconfig);
+    kubeconfig = NULL;
+
+    return rc;
+}
+
+int load_kube_config_buffer(char **pBasePath, sslConfig_t ** pSslConfig, list_t ** pApiKeys, const char *buffer)
+{
+    static char fname[] = "load_kube_config_buffer()";
+    int rc = 0;
+
+    kubeconfig_t *kubeconfig = kubeconfig_create();
```

Comment:
Stack allocate here to, as above.

Reply:
Same here. See https://github.com/kubernetes-client/c/pull/242#discussion_r1663546396